dali/operators/image/paste/multipaste.cc

```diff
@@ -0,0 +1,224 @@
+  // as a reference has to be set, but we do not use it anywhere
+  const auto &out_idx = ws.template InputRef<CPUBackend>(input_out_ids_ ? 5 : 1);
+
+  const auto &output = ws.template OutputRef<CPUBackend>(0);
+  output_desc.resize(1);
+
+  const int n_samples = in_anchors.shape().num_samples();
+
+  for (int i = 0; i < n_samples; i++) {
+    const int n_paste = in_anchors[i].shape()[0];
+
+    DALI_ENFORCE(in_anchors[i].shape()[0] == n_paste,
+                 "in_anchors must be same length as in_idx");
+    DALI_ENFORCE(in_shapes[i].shape()[0] == n_paste,
+                 "in_shapes must be same length as in_idx");
+    DALI_ENFORCE(out_anchors[i].shape()[0] == n_paste,
+                 "out_anchors must be same length as in_idx");
+
+    if (input_out_ids_) {
+      DALI_ENFORCE(out_idx[i].shape()[0] == n_paste,
+                   "out_idx must be same length as in_idx");
+    }
+  }
+
+  TYPE_SWITCH(images.type().id(), type2id, InputType, (uint8_t, int16_t, int32_t, float), (
+      TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
+          {
+            using Kernel = TheKernel<OutputType, InputType>;
+            kernel_manager_.Initialize<Kernel>();
+            TensorListShape<> sh = images.shape();
+            TensorListShape<> shapes(sh.num_samples(), 3);
+            for (int i = 0; i < sh.num_samples(); i++) {
+              const TensorShape<> &out_sh =
+                dali::TensorShape<>(output_height_[i], output_width_[i], sh[i][2]);
+              shapes.set_tensor_shape(i, out_sh);
+            }
+
+            TypeInfo type;
+            type.SetType<OutputType>(output_type_);
+            output_desc[0] = {shapes, type};
+          }
+      ), DALI_FAIL(make_string("Unsupported output type: ", output_type_)))  // NOLINT
+  ), DALI_FAIL(make_string("Unsupported input type: ", images.type().id())))  // NOLINT
+  return true;
+}
+
+
+void MultiPasteCpu::RunImpl(workspace_t<CPUBackend> &ws) {
+  const auto &images = ws.template InputRef<CPUBackend>(0);
+  const auto &in_idx = ws.template InputRef<CPUBackend>(1);
+  const auto &in_anchors = ws.template InputRef<CPUBackend>(2);
+  const auto &in_shapes = ws.template InputRef<CPUBackend>(3);
+  const auto &out_anchors = ws.template InputRef<CPUBackend>(4);
+  const auto &out_idx = ws.template InputRef<CPUBackend>(input_out_ids_ ? 5 : 1);
+
+  auto &output = ws.template OutputRef<CPUBackend>(0);
+
+  output.SetLayout(images.GetLayout());
+  auto out_shape = output.shape();
+
+  auto& tp = ws.GetThreadPool();
+
+  auto batch_size = in_idx.shape().num_samples();
+
+  TYPE_SWITCH(images.type().id(), type2id, InputType, (uint8_t, int16_t, int32_t, float), (
+      TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
+          {
+            using Kernel = TheKernel<OutputType, InputType>;
+            if (no_intersections_) {
+              for (int i = 0; i < batch_size; i++) {
```

Comment:
This should iterate over _output_ batch, not input - see my comments about out_ids

Reply:
Changed